core/templates/pages/exploration-editor-page/translation-tab/state-translation-editor/state-translation-editor.component.spec.ts

```diff
@@ -372,6 +388,11 @@ describe('State Translation Editor Component', function() {
       ctrl.$onInit();
     }));
 
+    afterEach(() => {
+      ctrl.$onDestroy();
```

Comment:
Just wanted to check, do we need to do this for all spec files where use $onDestroy() to unsubscribe event subscriptions?

Reply:
These functions ($onInit, $onDestroy in AJS and ngOnInit and ngOnDestroy in Angular 2+) are called by the DOM when the component gets initialized and destroyed. However, since we are invoking them in tests, we do manually call them. I'm simply calling them to complete 100% coverage, since there is just an rxjs unsubscribe inside. 

As for having a common method, I don't think so. Because we don't particularly need an $onDestroy/ngOnDestroy for every component, only components where we explicitly wanted to do something when the component is destroyed (like unsubscribing the observable in this case)

Reply:
I see, thanks for explaining. One follow-up question, if $onDestroy were not added in this test, would rxjs "subscribe" remain active until all the tests have completed? If so, can it lead to flaky tests?

Reply:
No, it's not going to lead to flaky tests, since we're not actually using the real emitters in the tests, just using mock emitters.

Reply:
Ah I see. Makes sense, thanks!